src/fourierdomain/nuft/nfft_alg.jl

```diff
@@ -107,6 +107,44 @@ end
     return nothing
 end
 
+function forward(
```

Comment:
```suggestion
function EnzymeRules.forward(
```
This is why the rule isn't being called and you are getting the indefinite number of fields error.